packages/core/src/workflows/workflow.ts

```diff
@@ -238,6 +238,10 @@ export class Workflow<
     const runId = crypto.randomUUID();
     this.#runId = runId;
 
+    if (!this.#mastra) {
+      throw new Error('Mastra not initialized, please register the workflow with a Mastra instance.');
```

Comment:
@abhiaiyer91 we need at least storage with suspend/resume so we need it to be registered or storage needs to be passed in. 

So this makes sense to me but it means, we cannot use workflow as a standalone class anymore. We could throw this error in resume or suspend call as well.